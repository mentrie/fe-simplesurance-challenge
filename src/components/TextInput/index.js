import React from 'react'
import PropTypes from 'prop-types';
import TextArea from 'react-autosize-textarea';

const TextInput = ({value, type, error, onChange, question, className="", ...restProps}) => {
  return (
    <div className="textinput">
      <label className="textinput__label">{value.text}</label>
      { type === 'string'?
        <TextArea 
          value={value.reply}
          onChange={onChange}
          placeholder={value.reply} 
          className={`${className} ${error? "has-error": ""}`}
          rows={5}
          {...restProps}
        /> :
        <input
          type={type}
          value={value.reply}
          onChange={onChange}
          placeholder={value.reply} 
          className={`${className} ${error? "has-error": ""}`}
          {...restProps}
        />
      }
      {error && <p classNam="error">{error}</p>}
    </div>
  )
}

TextInput.propTypes = {
  /// Add propTypes validation here
}

export default TextInput;
