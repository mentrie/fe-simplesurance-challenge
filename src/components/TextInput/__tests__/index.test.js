import React from 'react'
import TextInput from '../index';

test('should render component', () => {
  const props = {
    value: {
      text: '',
      reply: ''
    },
    onChange: jest.fn()
  }
  const component = shallow(<TextInput {...props} />);
  expect(component).toMatchSnapshot();
});

test('It should render TextArea if question type is string', () => {
  const props = {
    value: {
      text: '',
      reply: ''
    },
    type: 'string',
    onChange: jest.fn()
  }

  const component = shallow(<TextInput {...props} />);
  expect(component).toMatchSnapshot();
  expect(component.find('TextareaAutosize').length).toBe(1);
});

test('It should render input tag if question type is not a string', () => {
  const props = {
    value: {
      text: '',
      reply: ''
    },
    type: 'date',
    onChange: jest.fn()
  }

  const component = shallow(<TextInput {...props} />);
  expect(component).toMatchSnapshot();
  expect(component.find('input').length).toBe(1);
});

