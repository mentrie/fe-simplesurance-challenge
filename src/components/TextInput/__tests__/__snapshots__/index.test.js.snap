// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`It should render TextArea if question type is string 1`] = `
<div
  className="textinput"
>
  <label
    className="textinput__label"
  />
  <TextareaAutosize
    className=" "
    onChange={[Function]}
    placeholder=""
    required={true}
    rows={5}
    value=""
  />
</div>
`;

exports[`It should render input tag if question type is not a string 1`] = `
<div
  className="textinput"
>
  <label
    className="textinput__label"
  />
  <input
    className=" "
    onChange={[Function]}
    placeholder=""
    required={true}
    type="date"
    value=""
  />
</div>
`;

exports[`should render component 1`] = `
<div
  className="textinput"
>
  <label
    className="textinput__label"
  />
  <input
    className=" "
    placeholder=""
    required={true}
    value=""
  />
</div>
`;
